casper/src/main/scala/coop/rchain/casper/BlockCreator.scala

```diff
@@ -66,8 +79,11 @@ object BlockCreator {
               Tools.rng(invalidBlockHash.toByteArray)
             )
         )
-        deploys          <- extractDeploys(dag, parentMetadatas, maxBlockNumber, expirationThreshold)
-        parents          <- parentMetadatas.toList.traverse(p => ProtoUtil.getBlock(p.blockHash))
+        deploys <- extractDeploys(dag, parentMetadatas, maxBlockNumber, expirationThreshold)
+        parents <- parentMetadatas.toList.traverse(p => ProtoUtil.getBlock(p.blockHash))
+        isActive = parents
+          .traverse(b => isActiveValidator(b, runtimeManager, validatorIdentity))
+          .map(_.foldRight(true)(_ && _))
```

Comment:
Please add some comments about the logic how to decide if it's active validator.